src/install.c

```diff
@@ -1158,6 +1158,99 @@ static gboolean launch_and_wait_default_handler(RaucInstallArgs *args, gchar* bu
 	return TRUE;
 }
 
+static gchar* get_uptime(void)
+{
+	g_autofree gchar *contents = NULL;
+	g_autoptr(GError) ierror = NULL;
+	g_auto(GStrv) uptime = NULL;
+	gchar *total_time = NULL;
+
+	if (!g_file_get_contents("/proc/uptime", &contents, NULL, &ierror)) {
+		g_warning("Failed to get uptime: %s", ierror->message);
+		return NULL;
+	}
+
+	/* file contains two values and a newline, 'chomp' in-place and split then */
+	uptime = g_strsplit(g_strchomp(contents), " ", 2);
+
+	/* hand char referenced by uptime[0] over to total_time to prevent autofreeing */
+	total_time = uptime[0];
+	uptime[0] = NULL;
```

Comment:
Will uptime[1] be freed this way? `g_strfreev()` docs talk about a NULL-terminated array. Simply use `g_strdup()`.

Reply:
Ack